app/src/main/java/com/example/music_app/MusicAppApplication.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.music_app
+
+import android.app.Application
+import android.content.Context
+
+class MusicAppApplication: Application()  {
+
+    override fun onCreate() {
+        instance = this
+        super.onCreate()
+    }
+
+    companion object {
+        private var instance: MusicAppApplication? = null
```

Comment:
Using `lateinit` modifier is usually a bad practice, but in this case it could useful to avoid nullability of context depending app parts like data store